### PR TITLE
Add descriptions for queries

### PR DESCRIPTION
**Pull request by maarcingebala**
This PR adds descriptions to queries and query arguments. These changes were made when I was working on the initial version of API reference for our docs (https://github.com/mirumee/saleor-docs/pull/45). There is still a lot of descriptions to add but I'm planning on doing it in separate PRs)

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for e950d9efb35d3f4a2833dba5eb8221d934d577e9 (maarcingebala/saleor @ graphql-docs)
Base comparison is 5e4e4a4259a011d24b04ebd24c77c689de843fa1.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         38	         38	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         52	         52	             24

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         15	         15	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         18	         18	              8
```

</p>
</details>

File: saleor/graphql/account/schema.py [hunk 1]
```diff
@@ -83,39 +83,52 @@ class Meta:
...
+        ),
         description="Lookup a service account by ID.",
     )
 
     user = graphene.Field(
         User,
-        id=graphene.Argument(graphene.ID, required=True),
-        description="Lookup a user by ID.",
+        id=graphene.Argument(graphene.ID, description="ID of the user.", required=True),
+        description="Lookup an user by ID.",
```
Comment (NyanKiyoshi) [inline saleor/graphql/account/schema.py]: It is a user, not an :+1:

File: saleor/graphql/core/fields.py [hunk 2]
```diff
@@ -14,6 +14,38 @@
...
+    field.args["last"].description = "Returns the last n elements from the list."
+    field.args[
+        "before"
+    ].description = (
+        "Returns the elements in the list that come before the specified cursor."
+    )
+    field.args[
+        "after"
+    ].description = (
+        "Returns the elements in the list that come after the specified cursor."
```
Comment (NyanKiyoshi) [inline saleor/graphql/core/fields.py]: For consistency 

```suggestion
        "Return the elements in the list that come after the specified cursor."
```

File: saleor/graphql/core/fields.py [hunk 3]
```diff
@@ -14,6 +14,38 @@
...
+
+    By default Graphene's connection fields comes without description for pagination
+    arguments. This functions patches those fields to add the descriptions.
+    """
+    field.args["first"].description = "Returns the first n elements from the list."
+    field.args["last"].description = "Returns the last n elements from the list."
+    field.args[
+        "before"
+    ].description = (
+        "Returns the elements in the list that come before the specified cursor."
```
Comment (NyanKiyoshi) [inline saleor/graphql/core/fields.py]:
```suggestion
        "Return the elements in the list that come before the specified cursor."
```

File: saleor/graphql/core/fields.py [hunk 4]
```diff
@@ -14,6 +14,38 @@
...
 
 
+def patch_pagination_args(field: DjangoConnectionField):
+    """Add descriptions to pagination arguments in a connection field.
+
+    By default Graphene's connection fields comes without description for pagination
+    arguments. This functions patches those fields to add the descriptions.
+    """
+    field.args["first"].description = "Returns the first n elements from the list."
+    field.args["last"].description = "Returns the last n elements from the list."
```
Comment (NyanKiyoshi) [inline saleor/graphql/core/fields.py]:
```suggestion
    field.args["last"].description = "Return the last n elements from the list."
```

File: saleor/graphql/core/fields.py [hunk 5]
```diff
@@ -14,6 +14,38 @@
 from .types.money import Money, TaxedMoney
 
 
+def patch_pagination_args(field: DjangoConnectionField):
+    """Add descriptions to pagination arguments in a connection field.
+
+    By default Graphene's connection fields comes without description for pagination
+    arguments. This functions patches those fields to add the descriptions.
+    """
+    field.args["first"].description = "Returns the first n elements from the list."
```
Comment (NyanKiyoshi) [inline saleor/graphql/core/fields.py]:
```suggestion
    field.args["first"].description = "Return the first n elements from the list."
```

**Pull request merged**